javascript/packages/fury/lib/meta/MetaString.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+class MetaString {
+  static LOWER_SPECIAL = 5;
+  static LOWER_UPPER_DIGIT_SPECIAL = 6;
+
+  // Encode function based on the specified bits per character
+  static encode(str: string, bitsPerChar: number): Uint8Array {
+    const totalBits = str.length * bitsPerChar + 1;
```

Comment:
In Javascript, The length of emoji is greater than 1.
```Javascript
("😁").length === 2
```
We always need to be careful, it often causes bugs.